### PR TITLE
380: Idea to solve convergence problem

### PR DESCRIPTION
**Pull request by danielinteractive**
part of https://github.com/openpharma/mmrm/issues/380

### CODE REVIEW

**Comment by danielinteractive** [comment]
@yonicd maybe you are interested in reviewing this design PR - I think I solved the puzzle why SAS converges much better in the example

**Review by yonicd — COMMENTED** [review]
In general i think that the computation cost of the empirical covmat is small and would relax the constraint that this simulated data showed. 

To the question of MIVQUE0, i think that could be a longer term goal in the next release. It would give more flexibility for larger dimension designs (per [reference paper](https://support.sas.com/documentation/onlinedoc/v82/techreport_r105.pdf)) so we may hit against that problem anyways as more people use the package.

**Review by yonicd — COMMENTED** [review]
this [blogpost](https://yetanothermathprogrammingconsultant.blogspot.com/2019/11/sdp-model-imputing-covariance-matrix.html) may give a direction on a consistent way to resolve `NA` in the covmat

here is an R [package](http://cran.nexr.com/web/packages/sdpt3r/vignettes/sdpt3r.pdf) that has a solution for the "nearsest correlation matrix" problem

**Review by yonicd — APPROVED** [review]

**Comment by danielinteractive** [comment]
> this [blogpost](https://yetanothermathprogrammingconsultant.blogspot.com/2019/11/sdp-model-imputing-covariance-matrix.html) may give a direction on a consistent way to resolve `NA` in the covmat
> 
> here is an R [package](http://cran.nexr.com/web/packages/sdpt3r/vignettes/sdpt3r.pdf) that has similar solutions

Thanks Yoni - yeah I think we don't need those more complex solutions because we can start from the residuals data set

File: design/unstructured_optimization.Rmd [hunk 1]
```diff
@@ -354,12 +354,21 @@ sigma_emp3[na_inds[, c(2, 1), drop = FALSE]] <- sigma_emp3[na_inds]
...
+```{r, eval = FALSE}
 L_mat3 <- t(chol(sigma_emp3))
 ```
 
-Hm it is not ... So maybe let's go back to pivoting the Cholesky decomposition,
-fill in NAs there and then come up with a complete matrix?
+Force the PD using Matrix package
+
+```{r}
+sigma_emp3_pd <- Matrix::nearPD(sigma_emp3)$mat
```
Comment (danielinteractive) [inline design/unstructured_optimization.Rmd:364]: @yonicd ah this is cool! thanks

**Pull request merged**